sim-lib/src/lib.rs

```diff
@@ -656,14 +681,17 @@ async fn produce_simulation_results(
                 match output{
                     Some(simulation_output) => {
                         match simulation_output{
-                            SimulationOutput::PaymentSent(dispatched_payment) => {
-                                let source_node = nodes.get(&dispatched_payment.source).unwrap().clone();
-
-                                log::debug!("Tracking payment result for: {}.", hex::encode(dispatched_payment.hash.0));
+                            SimulationOutput::SendPaymentSuccess(payment) => {
+                                let source_node = nodes.get(&payment.source).unwrap().clone();
                                 set.spawn(track_payment_result(
-                                    source_node,results.clone(),simulation_output, shutdown.clone(),
+                                    source_node,results.clone(),payment, shutdown.clone(),
```

Comment:
Not sure why `fmt` is not catching this, but there are some spaces missing here.

Reply:
`fmt` is unreliable in `select` :(

Reply:
hm